test(unit): angular v11 example app and unit tests (the tour of heroes)

rfprod opened the pull request:
## Summary

- [x] add [the tour of heroes](https://angular.io/generated/live-examples/testing/stackblitz) app and specs to the example v11 app;
- [x] rewrite specs using jest;
- [x] remove unneeded comments from specs;
- [x] remove unneeded `console.error`, `console.log`;
- [x] address linter warnings;
- [x] remove source that are not present in the tour of heroes app;

## Does this PR introduce a breaking change?

- [X] No

## Other information

I testest v11 example app locally using the following commands
```bash
yarn test
yarn test-isolated
yarn test-esm
yarn test-esm-isolated
```

rfprod commented:
@ahnpnl review please. The original 'tour of heroes' unit tests appeared to be surprisingly messy as for me. I clean it up as much as I could but there are still some linter warnings present. Please let me know if you want me to fix it as well. Also, I enabled the `strict` compiler mode for the v11 example app to rewrite and debug tests.

ahnpnl commented:
> @ahnpnl review please. The original 'tour of heroes' unit tests appeared to be surprisingly messy as for me. I cleaned it up as much as I could but there are still some linter warnings present. Please let me know if you want me to fix linter warnings as well. Also, I enabled the `strict` compiler mode for the v11 example app to rewrite and debug tests.
> 
> One more thing, there were (and still are) a lot of repetitive commens that can make it hard to read the specs. The comments were present in the original specs in the 'tour of heroes' app. I cleaned up some but have not removed all of them. Please let me know if you want me to remove comments and commented code from the specs.

- I like the idea of enabled `strict` mode.
- I think we should remove those comments. We don't need those comments (including Google comment thing).
- I saw 36 lint warnings in the branch. I think we can fix all of those
- Would you please correct also any tests which use `console.error` if possible.

rfprod commented:
> > @ahnpnl review please. The original 'tour of heroes' unit tests appeared to be surprisingly messy as for me. I cleaned it up as much as I could but there are still some linter warnings present. Please let me know if you want me to fix linter warnings as well. Also, I enabled the `strict` compiler mode for the v11 example app to rewrite and debug tests.
> > One more thing, there were (and still are) a lot of repetitive commens that can make it hard to read the specs. The comments were present in the original specs in the 'tour of heroes' app. I cleaned up some but have not removed all of them. Please let me know if you want me to remove comments and commented code from the specs.
> 
> * I like the idea of enabled `strict` mode.
> * I think we should remove those comments. We don't need those comments (including Google comment thing).
> * I saw 36 lint warnings in the branch. I think we can fix all of those
> * Would you please correct also any tests which use `console.error` if possible.

@ahnpnl will do

rfprod commented:
@ahnpnl review please, I fixed everything.

rfprod commented:
@ahnpnl ok, not it's ready. I removed 2 unit tests that were skipped initially with a note that it's not working yet. These tests work with esm disabled, but don't work with esm enabled. In fact, testing routing on the unit level is a questionable topic. This type of tests are from the e2e realm.

If you really want unit tests for lazy routes, I can add it later. Need some more time to figure out why it does not work with esm modules.

ahnpnl commented:
For the router, I agree we don’t really need unit tests for it.

rfprod commented:
@ahnpnl so, is this PR good to merge? What do you think?
If it looks good, I can switching to other open draft PRs with unit tests (v12, v13, yarn workspace, monorepo).

ahnpnl commented on examples/example-app-v11/src/environments/environment.ts:
```diff
@@ -2,7 +2,9 @@
 // The list of file replacements can be found in `angular.json`.
 
-export const environment = {
+import { AppEnvironment } from '../app/configs/environment.config';
```
Only small thing is that the Angular example app doesn’t have this interface, we should remove this interface.

rfprod replied:
@ahnpnl removed all sources that are not present in the tour of heroes

ahnpnl commented:
Thanks!!!
